gen3-client/g3cmd/upload-multipart.go

```diff
@@ -0,0 +1,174 @@
+package g3cmd
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"sort"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/uc-cdis/gen3-client/gen3-client/commonUtils"
+	"github.com/uc-cdis/gen3-client/gen3-client/logs"
+	pb "gopkg.in/cheggaaa/pb.v1"
+)
+
+var multipartUploadLock sync.Mutex
+
+func retry(attempts int, filePath string, guid string, f func() error) (err error) {
+	for i := 0; ; i++ {
+		err = f()
+		if err == nil {
+			return
+		}
+
+		if i >= (attempts - 1) {
+			break
+		}
+
+		time.Sleep(GetWaitTime(i))
+
+		log.Println("Retrying after error: ", err)
+	}
+	return fmt.Errorf("After %d attempts, last error: %s", attempts, err)
+}
+
+func multipartUpload(uploadPath string, filePath string, numParallel int, includeSubDirName bool, retryCount int) error {
+	file, err := os.Open(filePath)
+	defer file.Close()
+	if err != nil {
+		logs.AddToFailedLogMap(filePath, "", "", retryCount, true, true)
```

Comment:
why is the parameter isMuted set to true here?

Reply:
there were complains about too much console output... I have planned for another update to add a `verbose` flag to control the amount of messages displaying. So i'm setting `isMuted` in multiple places here to temp reduce console output